### DISCUSSION TITLE
Can the image be transformed into perspective？

### DISCUSSION THREAD

**Discussion by banli17** [libvips/libvips#2057] [Q&A]
in imageMagick  -distort perspective

```
magick convert -virtual-pixel transparent -distort perspective "0,0,0,8 0,390,132,323 558,390,558,265 558,0,400,0" 1.png 2.png
```

According to the four vertices to transform, I see that affine api seems to only modify the xy displacement

**Comment by GavinJoyce** [comment]
@banli17 have you seen this approach with `mapim`? https://stackoverflow.com/a/50219715

**Comment by jcupitt** [comment]
I'm sorry @banli17, this completely dropped off my radar.

As @GavinJoyce says, you can transform with any function using `mapim`. There was a blog post a while ago about it:

http://libvips.blogspot.com/2015/11/fancy-transforms.html
